packages/core/src/sync-store/index.ts

```diff
@@ -566,64 +523,155 @@ export const createSyncStore = ({
+            "=",
+            "0x1",
           ),
         )
-        .where(ksql`${ksql.ref("callTraces.error")} IS NULL`)
+        .$if(filter.fromBlock !== undefined, (qb) =>
+          qb.where("blockNumber", ">=", filter.fromBlock!.toString()),
+        )
+        .$if(filter.toBlock !== undefined, (qb) =>
+          qb.where("blockNumber", "<=", filter.toBlock!.toString()),
+        );
+
+    const transferSQL = (
+      filter: TransferFilter,
+      db: Kysely<PonderSyncSchema>,
+      index: number,
+    ) =>
+      db
+        .selectFrom("traces")
+        .select([
+          ksql.raw(`'${index}'`).as("filterIndex"),
+          "checkpoint",
+          "chainId",
+          "blockHash",
+          "transactionHash",
+          ksql`null`.as("logId"),
+          "id as traceId",
+        ])
+        .where("chainId", "=", filter.chainId)
         .$call((qb) => addressSQL(qb as any, filter.fromAddress, "from"))
-        .$call((qb) => addressSQL(qb, filter.toAddress, "to"))
-        .where("blockNumber", ">=", filter.fromBlock.toString())
+        .$call((qb) => addressSQL(qb as any, filter.toAddress, "to"))
+        .where("value", ">", "0")
+        .$if(filter.includeReverted === false, (qb) =>
+          qb.where("error", "is", null),
+        )
+        .$if(filter.fromBlock !== undefined, (qb) =>
+          qb.where("blockNumber", ">=", filter.fromBlock!.toString()),
+        )
         .$if(filter.toBlock !== undefined, (qb) =>
           qb.where("blockNumber", "<=", filter.toBlock!.toString()),
         );
 
-    const blockSQL = (
-      filter: BlockFilter,
+    const traceSQL = (
+      filter: TraceFilter,
       db: Kysely<PonderSyncSchema>,
       index: number,
     ) =>
       db
-        .selectFrom("blocks")
+        .selectFrom("traces")
         .select([
           ksql.raw(`'${index}'`).as("filterIndex"),
           "checkpoint",
           "chainId",
-          "hash as blockHash",
-          ksql`null`.as("transactionHash"),
+          "blockHash",
+          "transactionHash",
           ksql`null`.as("logId"),
-          ksql`null`.as("callTraceId"),
+          "id as traceId",
         ])
         .where("chainId", "=", filter.chainId)
-        .$if(filter !== undefined && filter.interval !== undefined, (qb) =>
-          qb.where(ksql`(number - ${filter.offset}) % ${filter.interval} = 0`),
+        .$call((qb) => addressSQL(qb as any, filter.fromAddress, "from"))
+        .$call((qb) => addressSQL(qb as any, filter.toAddress, "to"))
+        .$if(filter.includeReverted === false, (qb) =>
+          qb.where("error", "is", null),
+        )
+        .$if(filter.callType !== undefined, (qb) =>
+          qb.where("type", "=", filter.callType!),
+        )
+        .$if(filter.functionSelector !== undefined, (qb) => {
+          if (Array.isArray(filter.functionSelector)) {
+            return qb.where((eb) =>
+              eb.or(
+                (filter.functionSelector as Hex[]).map((fs) =>
```

Comment:
I wonder if we should add a `functionSelector` column + index to the database - I could see this `like` performing poorly on large tables.